src/ngx_http_lua_shdict.c

```diff
@@ -531,6 +581,14 @@ ngx_http_lua_shdict_get_helper(lua_State *L, int get_stale)
         lua_pushboolean(L, c ? 1 : 0);
         break;
 
+    case LUA_TTABLE:
+
+        ngx_shmtx_unlock(&ctx->shpool->mutex);
+
+        lua_pushnil(L);
+        lua_pushliteral(L, "wrongtype operation");
```

Comment:
I found another question now:
Can we use `shdict.set` on list type key even when key is existed and not expired?
It's not allow now in this PR implementing, but I tested redis, it's ok to set key value on existed list type key.